jscomps/LoginManagerPrompter.js

```diff
@@ -90,106 +175,93 @@ LoginManagerPromptFactory.prototype = {
-    if (prompt.channel) {
-      var httpChannel = prompt.channel.QueryInterface(Ci.nsIHttpChannel);
-      if (httpChannel) {
-        var windowId = httpChannel.topLevelContentWindowId;
-        if (windowId != cancelationCounter.id) {
-          // window has been reloaded or navigated, reset the counter
-          cancelationCounter = { count: 0, id: windowId };
-        }
-      }
-    }
-
     var self = this;
 
     var runnable = {
       cancel: false,
-      run : function() {
+      run() {
         var ok = false;
         if (!this.cancel) {
           try {
-            self.log("_doAsyncPrompt:run - performing the prompt for '" + hashKey + "'");
-            ok = prompter.promptAuth(prompt.channel,
-                                     prompt.level,
-                                     prompt.authInfo);
+            self.log(
+              "_doAsyncPrompt:run - performing the prompt for '" + hashKey + "'"
+            );
+            ok = prompter.promptAuth(
+              prompt.channel,
+              prompt.level,
+              prompt.authInfo
+            );
           } catch (e) {
-            if (e instanceof Components.Exception &&
-                e.result == Cr.NS_ERROR_NOT_AVAILABLE) {
-              self.log("_doAsyncPrompt:run bypassed, UI is not available in this context");
+            if (
+              e instanceof Components.Exception &&
+              e.result == Cr.NS_ERROR_NOT_AVAILABLE
+            ) {
+              self.log(
+                "_doAsyncPrompt:run bypassed, UI is not available in this context"
+              );
             } else {
-              Components.utils.reportError("LoginManagerPrompter: " +
-                                           "_doAsyncPrompt:run: " + e + "\n");
+              Cu.reportError(
+                "LoginManagerAuthPrompter: _doAsyncPrompt:run: " + e + "\n"
+              );
             }
           }
 
           delete self._asyncPrompts[hashKey];
           prompt.inProgress = false;
           self._asyncPromptInProgress = false;
-
-          if (ok) {
-            cancelationCounter.count = 0;
-          } else {
-            cancelationCounter.count++;
-          }
-          if (prompter._browser) {
-            prompter._browser.canceledAuthenticationPromptCounter = cancelationCounter;
-          }
         }
 
         for (var consumer of prompt.consumers) {
-          if (!consumer.callback)
+          if (!consumer.callback) {
             // Not having a callback means that consumer didn't provide it
             // or canceled the notification
             continue;
+          }
 
           self.log("Calling back to " + consumer.callback + " ok=" + ok);
           try {
             if (ok) {
-              consumer.callback.onAuthAvailable(consumer.context, prompt.authInfo);
+              consumer.callback.onAuthAvailable(
+                consumer.context,
+                prompt.authInfo
+              );
```

Comment:
I'm fine having these kind of changes here but these do not relate to "Update to new nsILoginManagerPrompter interface" commit message.

Reply:
Yes, you're right, the commit message was all wrong here, thanks for picking it up. I've changed it so hopefully it's clearer what's going on now.